motoman_driver/src/industrial_robot_client/joint_trajectory_action.cpp

```diff
@@ -614,9 +637,19 @@ bool JointTrajectoryAction::withinGoalConstraints(
   {
     int last_point = traj.points.size() - 1;
 
+    std::vector<std::string> state_names;
+    std::vector<double> state_values;
+
+    // joint torso_joint_b2 is not in traj.joint_names, so the last group is ignored here.
```

Comment:
Yes, this is the same with the issue above.
we only use the first 3 groups value, and ignore the last torso_joint_b2 (**last_trajectory_state_map_.size()** ).